extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSource.java

```diff
@@ -122,8 +122,12 @@ public long size() {
 
         @Override
         public InputStream openStream() {
-            var request = GetObjectRequest.builder().key(keyName).bucket(bucketName).build();
-            return client.getObject(request);
+            try {
+                var request = GetObjectRequest.builder().key(keyName).bucket(bucketName).build();
+                return client.getObject(request);
+            } catch (Exception e) {
+                throw new RuntimeException(String.format("Fail downloading object from the source bucket: %s", e.getMessage()), e);
```

Comment:
please throw an `EdcException`, and either use static imports for the `format` call, or - even better - leverage `"xyzabc".formatted(...)`

Reply:
Rethrown exceptions removed